src/twisted/conch/ssh/transport.py

```diff
@@ -638,20 +638,39 @@ def sendPacket(self, messageType, payload):
                 payload
             ) + self.outgoingCompression.flush(2)
         bs = self.currentEncryptions.encBlockSize
-        # 4 for the packet length and 1 for the padding length
-        totalSize = 5 + len(payload)
+
+        isGCM = self.currentEncryptions.outCipType.endswith(b"-gcm@openssh.com")
```

Comment:
This really seems like the sort of thing which should be computed ahead of time and cached, not established via a string comparison every time `sendPacket` is called?

Reply:
(Relatedly maybe we should make some codspeed benchmarks for Conch)

Reply:
True. I will first work on #12186 as that PR already has some refactoring for this.
Once ETM is merged we can look into this PR